File: Models/Animal.cs

using System;

namespace FarmManagementSystem
{
    /// <summary>
    /// Model class representing an animal on the farm
    /// </summary>
    public class Animal
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public int Age { get; set; }
        public string Species { get; set; }
        public string Description { get; set; }
        public DateTime DateAdded { get; set; }
        
        public override string ToString()
        {
            return $"{Name} - {Species}, {Age} years old (ID: {Id})";
        }
    }
}

File: src/models/Animal.ts

// Animal model
export interface Animal {
  id: string;
  name: string;
  age: number;
  species: string;
  description?: string;
  dateAdded: string;
}

// Initial animals data
export const initialAnimals: Animal[] = [
  {
    id: '1',
    name: 'Bessie',
    age: 5,
    species: 'Cow',
    description: 'A friendly dairy cow with black and white spots',
    dateAdded: new Date().toISOString()
  },
  {
    id: '2',
    name: 'Wilbur',
    age: 2,
    species: 'Pig',
    description: 'A clever and friendly pig',
    dateAdded: new Date().toISOString()
  },
  {
    id: '3',
    name: 'Clucky',
    age: 1,
    species: 'Chicken',
    description: 'A reliable egg-laying hen',
    dateAdded: new Date().toISOString()
  }
];
